sys/kern/kern_exec.c

```diff
@@ -1184,19 +1184,19 @@ get_argenv_ptr(void * __capability *arrayp, void * __capability *ptrp)
 		*ptrp = __USER_CAP_STR((void *)(uintptr_t)ptr32);
 	} else
 #endif
-#if __has_feature(capabilities)
-	if (SV_CURPROC_FLAG(SV_CHERI)) {
-		if (fuecap(array, &ptr_c) == -1)
+#ifdef COMPAT_FREEBSD64
+	if (SV_CURPROC_FLAG(SV_LP64 | SV_CHERI) == SV_LP64) {
```

Comment:
I've often spelled this `!SV_CURPROC_FLAG(SV_CHERI)` and relied on the `SV_ILP32` case above to cover the other case, but I think I prefer this.